lib/policies/proxy/round-robin.js

```diff
@@ -1,21 +1,24 @@
+const url = require('url');
 const logger = require('../../logger').gateway;
 
 module.exports = class RoundRobin {
-  constructor (proxy, endpoints) {
+  constructor (proxy, proxyOptions, endpoints) {
     this._proxy = proxy;
 
+    this.proxyOptions = proxyOptions;
     this.endpoints = endpoints;
     this.endpointIndex = 0;
     this.endpointMaxIndex = this.endpoints.length - 1;
   }
 
   proxy (req, res) {
-    const options = {
-      target: this.nextTarget()
-    };
+    const nextTarget = this.nextTarget();
+    const parsedEndpointUrl = url.parse(nextTarget);
 
-    logger.debug(`proxying to ${options.target}, ${req.method} ${req.url}`);
-    return this._proxy.web(req, res, options);
+    this.proxyOptions.target = Object.assign(this.proxyOptions.target, parsedEndpointUrl);
```

Comment:
Why not simply use the equal operator?
`this.proxyOptions.target = parsedEndpointUrl`

Reply:
The key, cert and ca certificate properties exist in the passed proxyOptions, which is mapped to this.proxyOptions. Your suggestion would remove the SSL configuration passed in.